velox/functions/lib/Re2Functions.cpp

```diff
@@ -433,9 +433,13 @@ class OptimizedLike final : public VectorFunction {
       size_t reducedPatternLength) {
     switch (P) {
       case PatternKind::kExactlyN:
-        return input.size() == reducedPatternLength;
+        return stringCore::approxLengthUnicode(
```

Comment:
This is expensive. A large subset of the workload is all-ASCII. Can we check if input is ASCII and apply new logic only if it is not?

See https://facebookincubator.github.io/velox/develop/scalar-functions.html#all-ascii-fast-path

Reply:
The link shows `callAscii` in scalar interface, currently LIKE is implemented by extending `VectorFunction`, is it there a similar thing there?

Reply:
See isAsciiArgs velox/expression/SimpleFunctionAdapter.h 

Something similar can be used in a custom VectorFunction.

```
  /// Return true if all VARCHAR arguments are all-ASCII for the specified rows.
  static bool isAsciiArgs(
      const SelectivityVector& rows,
      const std::vector<VectorPtr>& args) {
    for (auto& arg : args) {
      if (arg->type()->isVarchar()) {
        auto stringArg = arg->asUnchecked<SimpleVector<StringView>>();
        auto isAscii = stringArg->isAscii(rows);
        if (!isAscii.has_value() || !*isAscii) {
          return false;
        }
      }
    }
    return true;
  }
```